src/reducers/photosReducer.js

```diff
@@ -212,120 +259,142 @@ export default function reducer(
-
-      updatedPhotosImageHashes = updatedPhotos.map(
-        photo => photo.image_hash
-      );
+      updatedPhotoDetails = action.payload.updatedPhotos;
+      newPhotoDetails = { ...state.photoDetails };
 
-      updatedPhotos.forEach(photo => {
-        newPhotos[[photo.image_hash]] = photo;
+      updatedPhotoDetails.forEach((photoDetails) => {
+        newPhotoDetails[photoDetails.image_hash] = photoDetails;
       });
 
-      var newPublicPhotos = [...state.publicPhotos];
-
-      if (valPublic) {
-        updatedPhotos.forEach(photo => {
-          newPublicPhotos.push(photo);
-        });
-      } else {
-        newPublicPhotos = newPublicPhotos.filter(photo => {
-          if (updatedPhotosImageHashes.includes(photo.image_hash)) {
-            return false;
-          } else {
-            return true;
-          }
-        });
-      }
-
       return {
         ...state,
-        photoDetails: newPhotos,
-        publicPhotos: newPublicPhotos
+        photoDetails: newPhotoDetails,
       };
     }
 
-    case "SET_PHOTOS_FAVORITE_FULFILLED": {
-      var valFavorite = action.payload.favorite;
-      updatedPhotos = action.payload.updatedPhotos;
-      newPhotos = { ...state.photoDetails };
-
-      updatedPhotosImageHashes = updatedPhotos.map(
-        photo => photo.image_hash
-      );
-
-      updatedPhotos.forEach(photo => {
-        newPhotos[[photo.image_hash]] = photo;
+    case SET_PHOTOS_FAVORITE_FULFILLED: {
+      updatedPhotoDetails = action.payload.updatedPhotos;
+      newPhotoDetails = { ...state.photoDetails };
+      newPhotosGroupedByDate = [...state.photosGroupedByDate];
+      newPhotosFlat = [...state.photosFlat];
+
+      updatedPhotoDetails.forEach((photoDetails) => {
+        newPhotoDetails[photoDetails.image_hash] = photoDetails;
+
+        newPhotosFlat = newPhotosFlat.map((photo) =>
+          photo.id === photoDetails.image_hash
+            ? { ...photo, rating: photoDetails.rating }
+            : photo
+        );
+        newPhotosGroupedByDate = newPhotosGroupedByDate.map((group) =>
+          // Create a new group object if the photo exists in its items (don't mutate).
+          group.items.find((photo) => photo.id !== photoDetails.image_hash)
+            ? group
+            : {
+                ...group,
+                items: group.items.map((item) =>
+                  item.id !== photoDetails.image_hash
+                    ? item
+                    : {
+                        ...item,
+                        rating: photoDetails.rating,
+                      }
+                ),
+              }
+        );
+
+        /**
+         * I'm not sure what the best behaviour would be when unfavoriting
+         * a photo when in the favorites view.
+         *
+         * Alternatives:
+         * 1. Let the unfavorited photo remain in the view (but turn off the star icon).
+         * 2. Remove the unfavorited photo from the view and close the lightbox.
```

Comment:
I think I prefer the second alternative too.
To close the lightbox, you have to change the state in PhotoListView of "lightboxShow"

https://github.com/LibrePhotos/librephotos-frontend/blob/263e69f063d1b45314dfaba0887a8f86ceabf0f3/src/components/photolist/PhotoListView.js#L41

Reply:
Right. Do you have a suggestion of how to accomplish that? I want able to figure out a good way...

Reply:
Code updated to close lightbox when unfavoriting in Favorited view.